code/game/machinery/computer/ai_core.dm

```diff
@@ -214,7 +214,7 @@ That prevents a few funky behaviors.
 							if (ticker.mode.name == "AI malfunction")
 								var/datum/game_mode/malfunction/malf = ticker.mode
 								for (var/datum/mind/malfai in malf.malf_ai)
-									if (T.mind == malfai)
+									if (T.mind == malfai && ticker.mode:malf_mode_declared)
```

Comment:
It's actually already typecasted for him right above it.

Should just be malf.malf_mode_declared.